www/source/javascripts/community.js

```diff
@@ -1,19 +1,128 @@
-var community_events = function(data) {
-    if ($(".community-events")) {
-        if (Array.isArray(data)) {
-            data.forEach(function(e) {
-                var ev = "<p>";
-                ev += e.title;
-                ev += "<br>";
-                ev += e.link;
-                ev += "<br>";
-                ev += e.pub_date;
-                ev += "<br>";
-                ev += e.description;
-                ev += "</p>";
-
-                $(".community-events").append(ev);
-            });
-        }
+$(function() {
+  var htmlDecode = function(input) {
+    var doc = new DOMParser().parseFromString(input, "text/html");
+    return doc.documentElement.textContent;
+  }
+
+  var makeEvent = function(e) {
+    if (e === undefined) {
```

Comment:
So, the name `makeEvent` here might be misleading.  I'm referring to community events that come back in the JSON feed, not javascript events.  The reason the check for undefined is necessary is because of the two column layout we're using.  Further down in the JS file, where I do this:

```javascript
var first = makeEvent(data[i]);
var second = makeEvent(data[i+1]);

row.append(first);
if (second) {
    row.append(second);
}
```
I'm looping through the events two at a time, in order to display two events side by side.  If there are an odd number of events, then the second event shouldn't be shown.  In that case, `data[i+1]` won't exist, and it will return `undefined`, so we end up passing `undefined` to `makeEvent`, and if that happens, we want to bail immediately and return `undefined` back because doing so means that it's easy to check whether we want to append the results of that function to the DOM.

Hopefully that clarifies what my intent was there.  Is there a better way to accomplish that goal?